aten/src/ATen/native/cudnn/MHA.cpp

```diff
@@ -412,6 +542,96 @@ void run_cudnn_SDP_fprop(
   mhagraphcache.update(key, graph_and_tensors_values);
 }
 
+void run_cudnn_SDP_bprop(
+    int64_t b,
+    int64_t h,
+    int64_t s_q,
+    int64_t s_kv,
+    int64_t d,
+    float scaling_factor,
+    bool is_causal,
+    float dropout_probability,
+    const Tensor& q,
+    const Tensor& k,
+    const Tensor& v,
+    const Tensor& o,
+    const Tensor& dO,
+    const Tensor& softmaxstats,
+    Tensor& dQ,
+    Tensor& dK,
+    Tensor& dV,
+    const Tensor& dropoutseed,
+    const Tensor& dropoutoffset) {
+   cudnnHandle_t handle = getCudnnHandle();
+   auto key = MHACacheKeyWrapper(
+       b,
+       h,
+       s_q,
+       s_kv,
+       d,
+       q,
+       k,
+       v,
+       dropout_probability,
+       is_causal,
+       true);
+   auto graph_and_tensors_backward_ptr = mhagraphbackwardcache.find(key); 
+   graph_and_tensors_backward graph_and_tensors_backward_values;
+   if (graph_and_tensors_backward_ptr) {
+     graph_and_tensors_backward_values = *graph_and_tensors_backward_ptr;
+   } else {
+     graph_and_tensors_backward_values = build_graph_and_tensors_backward(
+       b,
+       h,
+       s_q,
+       s_kv,
+       d,
+       scaling_factor,
+       is_causal,
+       dropout_probability,
+       q,
+       k,
+       v,
+       o,
+       dO,
+       softmaxstats,
+       dQ,
+       dK,
+       dV,
+       dropoutseed,
+       dropoutoffset,
+       handle,
+       key.pod);
+   }
+   auto [mha_graph, Q, K, V, attn_scale, Seed, Offset, O, Do, Stats, Dq, Dk, Dv] = graph_and_tensors_backward_values;
+   std::unordered_map<std::shared_ptr<fe::graph::Tensor_attributes>, void*> variant_pack = {
+    // inputs
+   {Q, q.data_ptr()},
```

Comment:
shouldn't some of these be `const_data_ptr`?

Reply:
Checked this again and it currently seems to be a limitation of cuDNN's `variantpack`, which only accepts `void *` pointers.